pygmt/src/grdfill.py

```diff
@@ -125,14 +125,16 @@ def grdfill(
     r"""
     Interpolate across holes in a grid.
 
+    .. note::
+
+       Wraps the GMT module :gmt-docs:`grdfill <grdfill.html>`.
+
     Read a grid that presumably has unfilled holes that the user wants to fill in some
     fashion. Holes are identified by NaN values but this criteria can be changed via the
     ``hole`` parameter. There are several different algorithms that can be used to
     replace the hole values. If no holes are found the original unchanged grid is
     returned.
 
-    Full option list at :gmt-docs:`grdfill.html`.
```

Comment:
The `note::` admonition stands out a bit too much. Perhaps just reword `Full option list at ...` to `Full GMT docs at ...`, similar to @ezevazquez's suggestion at https://github.com/GenericMappingTools/pygmt/issues/3881#issuecomment-2843105418?

Reply:
> The `note::` admonition stands out a bit too much.

I agree. 

> Perhaps just reword `Full option list at ...` to `Full GMT docs at ...`

I prefer to explicitly mention the name of the GMT module, since the function/method names may differ from the module names. What about the one below?

![image](https://github.com/user-attachments/assets/d4b48d8e-39f2-473c-ae18-c8f06302dea2)

Reply:
> since the function/method names may differ from the module names

I thought we usually use the same method name as upstream GMT? Different ones I can find are:
- https://www.pygmt.org/v0.15.0/api/generated/pygmt.Figure.set_panel.html (upstream uses `gmt subplot set`), and that page is missing the upstream doc link btw.
- https://www.pygmt.org/dev/api/generated/pygmt.which.html#pygmt.which (upstream uses `gmtwhich`)

The names seem to follow closely enough that we don't need to repeat the name really.

Reply:
We have methods like `hlines`/`vlines` and will have methods like `scatter` (#scatter), `choropleth` (#2798), which wraps `plot`. I guess we still need to link to `plot` in these methods.

Reply:
> We have methods like `hlines`/`vlines` and will have methods like `scatter` (#3602 ) and `choropleth` (#2798), which wrap `plot`. I guess we still need to link to `plot` in these methods.

For those convenience methods, then yes, it makes sense to use "Wraps GMT module ``scatter``, with full docs at :gmt-docs:`plot.html`.". But for most other ones, probably ok to use just "Full GMT docs at ..."?

Reply:
OK. I've changed it to "Full GMT docs at ...".